numpy/core/src/multiarray/mapping.c

```diff
@@ -252,7 +252,23 @@ unpack_indices(PyObject *index, PyObject **result, npy_intp result_n)
 }
 
 /**
- * Prepare an npy_index_object from the python slicing object.
+ * If arg is a callable, apply it to self.
+ *
+ * @returns arg, arg(self), or NULL. NULL is returned if arg(self) fails
+ *      (if the callable takes more than a single argument, for example).
+ */
+NPY_NO_EXPORT PyObject *
+maybe_apply_callable(PyArrayObject *self, PyObject *arg)
+{
```

Comment:
This could be a static function. That might help the compiler inline it.

Reply:
I see, do I replace `NPY_NO_EXPORT` with `static`? Or would line 260 start with `static NPY_NO_EXPORT ...`?

Reply:
changed it to `static NPY_INLINE` in the latest version, modeled after `unpack_indices`